src/extensionConflicts.ts

```diff
@@ -0,0 +1,69 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Red Hat, Inc. All rights reserved..
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+import { commands, Extension, extensions, window } from 'vscode';
+
+// A set of VSCode extension ID's that conflict with our extension
+const conflictingIDs = new Set([
+    'haaaad.ansible',
+    'sysninja.vscode-ansible-mod',
+    'tomaciazek.ansible',
+    'vscoss.vscode-ansible',
+    'zbr.vscode-ansible',
+]);
+
+// A set of VSCode extension ID's that are currently uninstalling
+const uninstallingIDs = new Set();
+
+/**
+ * Get all of the installed extensions that currently conflict with VSCode-YAML
+ */
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export function getConflictingExtensions(): Extension<any>[] {
+  const conflictingExtensions: Extension<any>[] = [];
+  conflictingIDs.forEach((extension) => {
+    const ext = extensions.getExtension(extension);
+    if (ext && !uninstallingIDs.has(ext.id)) {
+      conflictingExtensions.push(ext);
+    }
+  });
+  return conflictingExtensions;
+}
+
+/**
+ * Display the uninstall conflicting extension notification if there are any conflicting extensions currently installed
+ */
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export function showUninstallConflictsNotification(conflictingExts: Extension<any>[]): void {
+  // Add all available conflicting extensions to the uninstalling IDs map
+  for (const extIndex in conflictingExts) {
```

Comment:
It's more straightforward to just use a "for ... of ..." loop
```suggestion
  for (let ext of conflictingExts) {
```